### PR TITLE
Category counts refresh in async task, PLATFORM-4181

### PR DESCRIPTION
**Pull request by abador**

### CODE REVIEW

**Review by Michal-Chatlas1221 — APPROVED** [review]

File: includes/WikiPage.php [hunk 1]
```diff
@@ -2774,102 +2774,14 @@ public function getAutoDeleteReason( &$hasHistory ) {
 	 * @param $deleted array The names of categories that were deleted
 	 */
 	public function updateCategoryCounts( $added, $deleted ) {
-		/* Wikia change begin - @author: wladek */
-		global $wgSkipCountForCategories;
+
+		$task = new \Wikia\Tasks\Tasks\RefreshCategoryCountsTask();
+		$task->setQueue( \Wikia\Tasks\Queues\CategoryCountsQueue::NAME );
+		$task->call( 'updateCounts',  $this->mTitle->getNamespace(), $added, $deleted );
+		$task->queue();
```
Comment (mszabo-wikia) [inline includes/WikiPage.php:2794]: We should skip queueing the task if no categories were added or deleted

**Review by Grunny — APPROVED** [review]

File: extensions/wikia/CategorySelect/CategorySelectController.class.php [hunk 2]
```diff
@@ -85,7 +85,7 @@ public function categories() {
 				}
 				$text = $this->wg->ContLang->convertHtml( $title->getText() );
 				$data[ $name ] = array(
-					'link' => Linker::link( $title, $text ),
+					'link' => Linker::link( $title, $text , [], [], [ 'known' ] ),
```
Comment (mszabo-wikia) [inline extensions/wikia/CategorySelect/CategorySelectController.class.php:88]: I'm not sure if this behavior is correct—the category should be redlinked if the page doesn't exist

Comment (abador) [reply extensions/wikia/CategorySelect/CategorySelectController.class.php:88]: At the moment of changing categories on the bottom part of the page there is lag. The moment the request is made those new categories don't exist. As far as I can tell categories are always created when someone adds them to an article

Comment (mszabo-wikia) [reply extensions/wikia/CategorySelect/CategorySelectController.class.php:88]: You can add a nonexistent category to the article and the category page can be visited and will show the article as a member, but that doesn't mean the category page itself exists as a wiki page—it must be created manually.

**Pull request merged**